plugins/org.eclipse.n4js/src/org/eclipse/n4js/external/LibraryManager.java

```diff
@@ -278,8 +278,8 @@ private void installDependenciesOfNPMs(IProgressMonitor monitor, List<LibraryCha
 			String versionRequested = Strings.emptyIfNull(reqestedNpm.getValue());
 			if (installedNpms.containsKey(name)) {
 				org.eclipse.emf.common.util.URI location = installedNpms.get(name).getKey();
-				String versionInstalled = installedNpms.get(name).getValue();
-				if (versionRequested.equals(Strings.emptyIfNull(versionInstalled))) {
+				String versionInstalled = Strings.emptyIfNull(installedNpms.get(name).getValue());
+				if (versionRequested.isEmpty() || versionRequested.equals(versionInstalled)) {
```

Comment:
That is improvement, but it would be even better to use `org.eclipse.n4js.utils.Version.compareTo(Version)`
While our implementation does not cover all SemVer cases, and npm extensions, it is better than string comparison.